client/my-sites/email/email-management/home/email-plan-subscription.jsx

```diff
@@ -13,6 +14,19 @@ import AutoRenewToggle from 'calypso/me/purchases/manage-purchase/auto-renew-tog
 import { withLocalizedMoment } from 'calypso/components/localized-moment';
 
 class EmailPlanSubscription extends React.Component {
+	isSubscriptionExpired() {
+		const { isLoadingPurchase, purchase } = this.props;
+
+		if ( isLoadingPurchase || ! purchase ) {
+			return false;
+		}
+
+		const todayTimestamp = new Date().setUTCHours( 0, 0, 0, 0 );
+		const expiryTimestamp = new Date( purchase.expiryDate ).getTime();
+
+		return todayTimestamp > expiryTimestamp;
```

Comment:
Determining if a subscription has expired seems like the sort of logic that could be used in some other places. @sirbrillig @nbloomf: could you confirm that we don't have already a helper function for this (I had a look to our code but couldn't find any but it's possible that I've missed something). Second, where would you recommend putting such a new helper? Maybe `client/lib/purchases/utils.ts`?

Reply:
I was thinking about extracting this, but it wasn't clear to me where that logic should go either. :-/

I'd love to refactor this somewhere more general (or use an existing function!)